test/metabase/driver/common/parameters/temporal_units_test.clj

```diff
@@ -29,26 +29,39 @@
                                   :template-tags {"time-unit" {:name "id"
                                                                :display-name "id"
                                                                :type         :temporal-unit}}))
-            year-query (assoc native-query
-                              :parameters [{:type   :temporal-unit
-                                            :name   "time-unit"
-                                            :target [:variable [:template-tag "time-unit"]]
-                                            :value  "year"}])
-            month-query (assoc native-query
-                               :parameters [{:type   :temporal-unit
-                                             :name   "time-unit"
-                                             :target [:variable [:template-tag "time-unit"]]
-                                             :value  "month"}])]
-        (mt/with-native-query-testing-context year-query
-          (is (= [[#t "2019-01-01"]
-                  [#t "2018-01-01"]
-                  [#t "2019-01-01"]]
-                 (run-sample-query year-query))))
-        (mt/with-native-query-testing-context month-query
-          (is (= [[#t "2019-02-01"]
-                  [#t "2018-05-01"]
-                  [#t "2019-12-01"]]
-                 (run-sample-query month-query))))))))
+            parameterized-query (assoc native-query
+                                       :parameters [{:type   :temporal-unit
+                                                     :name   "time-unit"
+                                                     :target [:variable [:template-tag "time-unit"]]}])
+            date-types ["minute" "hour" "day" "week" "month" "quarter" "year"]
+            count-types ["second-of-minute" "minute-of-hour" "hour-of-day" "day-of-month"
+                         "day-of-year" "month-of-year" "quarter-of-year"]
+            ;; The original date is 2019-02-11T21:40:27.892
```

Comment:
nit: maybe pick a date that's not in the first quarter so that "quarter" and "year" are distinct?